### ISSUE TITLE
Excel memory problem

### ISSUE THREAD

**Issue by ShieldTrade** [cunnane/xloil#83]
Hi @cunnane ,

I experience an increase in Excel memory usage, which eventually crashes Excel, when I run the following code:

B2 = fetch_ohlcv(10000)
L2 = dummy_ind($B$2#)

copy and past L2 4 times. After 5 min you can see it in the task manager.

Thank you for your help and I wish you a very happy Christmas.

```
import asyncio
import xloil as xlo
import xloil.pandas
import pandas as pd
import random

"""
fetch time series every 2 seconds and trigger excel to make some calculation 
"""

@xlo.func()
async def fetch_ohlcv(steps)-> pd.DataFrame:
    l_steps = list(range(0,int(steps),1))
    random.seed(0)
    while True:
        l_open = [random.random() for _ in range(int(steps))]
        ohlcv = {
            'T': l_steps,
            'open' :l_open,
            'high':[i * 1.02 for i in l_open],
            'low':[i * 0.98 for i in l_open],
            'close':[i * 1.01 for i in l_open],
            'vol': [i * 10 for i in l_open]
                  }
        df = pd.DataFrame(ohlcv).set_index('T')
        yield df
        await asyncio.sleep(2)

@xlo.func()
def dummy_ind(df:pd.DataFrame):
    series = df['close'].multiply(-1) # <class 'pandas.core.series.Series'>
    # no diff if a list object is returned
    return series

@xlo.func()  # same behaviour as in dummy_ind()
def same_df(df:pd.DataFrame):
    return df
```

**Comment by andyvandy** [comment]
I think the root cause of the resource leakage may be related to issue https://github.com/cunnane/xloil/issues/81 

I think an unlimited amount of cached objects can pile up as long as the calcId remains unchanged. 
https://github.com/cunnane/xloil/blob/1199790cb5c42b24f431de33212bda3431185901/include/xloil/ObjectCache.h#L99-L109

**Comment by cunnane** [comment]
I've just released 0.18.5 which fixes this
